### PR TITLE
doc: changes to buildlinux

### PR DESCRIPTION
**Pull request by BIGBEASTISHANK**
In the docs the steps given to use d++ when build from source and located in `/usr/local/` was **incorrect** corrected it in this PR

## Documentation change checklist

- [x] My documentation changes follow the same style as the rest of the documentation and any examples follow the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR (via running `doxygen`, and testing examples).
- [x] I have not moved any existing pages or changed any existing URLs without strong justification as to why.
- [x] I have not generated content using AI or a desktop utility such as grammarly.

### CODE REVIEW

File: docpages/building/linux.md
Comment (Jaskowicz1) [inline docpages/building/linux.md:45]: Have you tried without chmod?

Comment (BIGBEASTISHANK) [reply docpages/building/linux.md:45]: > Have you tried without chmod?

Yes i did `ldd /usr/local/libdpp.so` and the output was `ldd: warning: you do not have execution permission for /usr/local/lib/libdpp.so.10.0.29'`  so yaa chmod is required

Comment (Jaskowicz1) [reply docpages/building/linux.md:45]: Weird, did you `sudo make install`?

Comment (BIGBEASTISHANK) [reply docpages/building/linux.md:45]: > Weird, did you `sudo make install`?

yes

Comment (Jaskowicz1) [reply docpages/building/linux.md:45]: Alrighty.

**Comment by Jaskowicz1** [comment]
I'll let someone else have a look at this, it seems off to me so I'm not too sure on it.

**Comment by BIGBEASTISHANK** [comment]
> I'll let someone else have a look at this, it seems off to me so I'm not too sure on it.

ok

**Comment by BIGBEASTISHANK** [comment]
> Also, if you've tested your PR then mark that checkbox.

I Have done that, any updates?

**Comment by Jaskowicz1** [comment]
> > Also, if you've tested your PR then mark that checkbox.
> 
> I Have done that, any updates?

We've told you before to be patient. It's been 16 hours since that message. PRs can take days to be reviewed, especially something like this that requires someone who knows Arch Linux to confirm if this is a good solution.

Please, be patient, we will approve when we can. We are not full-time PR reviewers as we have jobs and/or education to attend.

File: docpages/building/linux.md
Comment (braindigitalis) [reply docpages/building/linux.md:45]: are you sure you don't have a non standard umask?

File: docpages/building/linux.md
Comment (braindigitalis) [inline docpages/building/linux.md:63]: should be "required to specify"

**Review by braindigitalis — CHANGES_REQUESTED** [review]
as above

File: docpages/building/linux.md
Comment (Neko-Life) [inline docpages/building/linux.md:45]: the correct permission for libraries should be 755, it should be executable to be linkable iirc

Comment (Neko-Life) [reply docpages/building/linux.md:45]: actually i missed that second chmod, you can remove the second chmod and specify 755 directly in the command so it's cleaner

File: docpages/building/linux.md
Comment (braindigitalis) [reply docpages/building/linux.md:45]: this will need to be updated every time we make a release. this is a pain point. should set your umask correct before make install

Comment (BIGBEASTISHANK) [reply docpages/building/linux.md:45]: > are you sure you don't have a non standard umask?

I have standard umask
```bash
ishank@Beast ~> umask
0022
```
standard umask is 0022 and this is what i have

Comment (BIGBEASTISHANK) [reply docpages/building/linux.md:45]: > this will need to be updated every time we make a release. this is a pain point. should set your umask correct before make install

we are building from source which already takes a lot of command, and can not be updated automaticaly even if you release new versions, we need to build new releases from source and then adding these small like to set permission is not a long process and also this is the only solution (as far as i have researched for `/usr/local/`)

File: docpages/building/linux.md
Comment (BIGBEASTISHANK) [reply docpages/building/linux.md:63]: ooh i got it

File: docpages/building/linux.md
Comment (BIGBEASTISHANK) [reply docpages/building/linux.md:45]: Hmm, sorry my fault, also their are some mistake in naming file too,

File: docpages/building/linux.md
Comment (Jaskowicz1) [inline docpages/building/linux.md:55]: -L isn't used here anymore.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/building/linux.md
Comment (BIGBEASTISHANK) [reply docpages/building/linux.md]: also `-ldpp` but i fixed it

Comment (Jaskowicz1) [reply docpages/building/linux.md]: Why is `-ldpp` not required anymore? Is it because of `/usr/local/lib/libdpp.so`? If so, can you see about swapping it for `-ldpp`.

File: docpages/building/linux.md
Comment (Jaskowicz1) [inline docpages/building/linux.md:52]: I've also just realised this no longer exists for the Non-Arch Linux compile command (line 37). Please add the section about important flags for that.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/building/linux.md
Comment (Jaskowicz1) [reply docpages/building/linux.md:52]: image

You need the important flags for the section about compiling, where you aren't talking about arch linux.

Comment (Jaskowicz1) [reply docpages/building/linux.md:52]: You should have two, keep the one you currently have, just duplicate this section and put it where the image above shows. Then correct the text.

File: docpages/building/linux.md
Comment (BIGBEASTISHANK) [reply docpages/building/linux.md]: `-lpdd` is cant not be used because it will tell linker to find it in standard location which is `/usr/lib` thats why.

`/usr/local/lib/libdpp.so` adding this specify the path of library
and then
`-Wl,-rpath,/usr/local/lib` adding this specigy the path of directory
it is required for some werid reason i dont know, its not running with that, if we sont specify the path of directory then it wont work, we need to specify the library file as well as directory

File: docpages/building/linux.md
Comment (Jaskowicz1) [reply docpages/building/linux.md]: Yep perfect, thank you!

File: docpages/building/linux.md
Comment (BIGBEASTISHANK) [reply docpages/building/linux.md]: > But if the CMake original install location is `/usr/local/lib`, then `-ldpp` should look there?

no thats the problem, cmake install it in `/usr/local/lib` and `-ldpp` is set to standard location, which in other linux(s) the standard location is `/usr/local/lib` but in arch it is `/usr/lib`, cmake install it in the **specific** place and then that **specific** place is standard for debian and all not for arch

Comment (Jaskowicz1) [reply docpages/building/linux.md]: how strange, alrighty.

**Review by Jaskowicz1 — APPROVED** [review]
I'm happy with this. Granted, doing this over and over might be a pain but if it's what is required for Arch then so be it. We really need to get that AUR package updated soon.

**Comment by BIGBEASTISHANK** [comment]
> We really need to get that AUR package updated soon.

even if you update your aur packages then changes i made will be required for arch, because apprantly **brain** want it in local only, the dpp package in aur install it in `/usr/lib` so yaa, for build from source you need my changes, and i curretly have my `libdpp.so` in `/usr/local/lib/` and i am using my command that ong one and it is working perfectly for me

**Comment by BIGBEASTISHANK** [comment]
> > > We really need to get that AUR package updated soon.
> > 
> > 
> > even if you update your aur packages then changes i made will be required for arch, because apprantly **brain** want it in local only, the dpp package in aur install it in `/usr/lib` so yaa, for build from source you need my changes, and i curretly have my `libdpp.so` in `/usr/local/lib/` and i am using my command that ong one and it is working perfectly for me
> 
> Only source requires your changes, the AUR package would go in `/usr/lib/`. This has already been discussed.

yaa that only i am saying lol

**Comment by Neko-Life** [comment]
> I'm happy with this. Granted, doing this over and over might be a pain but if it's what is required for Arch then so be it. We really need to get that AUR package updated soon, just gotta wait for someone to take it over sadly.

it's only required for newbie, building stuff from source is common for arch users especially when you're ricing your OS :)

**Comment by BIGBEASTISHANK** [comment]
> building stuff from source is common for arch users especially when you're ricing your OS
Not quite true but yaa we need to take care of newbies too, we cant leave it

**Comment by BIGBEASTISHANK** [comment]
I guess my pr is ready to be merged, Brain do you have any other question or objection?

**Comment by braindigitalis** [comment]
user stated they are <13 on discord, as such according to github TOS they are not allowed on this service either. We can't report them here for what they said on discord but we can and should block them from the brainboxdotcc org.

**Pull request closed without merging**
